tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/BuildCompleteQueueWorker.cs

```diff
@@ -1,43 +1,46 @@
-using System;
+﻿using System;
 using System.Text.RegularExpressions;
+using System.Threading;
 using System.Threading.Tasks;
+using Azure.Storage.Queues;
 using Azure.Storage.Queues.Models;
 using Microsoft.ApplicationInsights;
 using Microsoft.ApplicationInsights.DataContracts;
-using Microsoft.Azure.WebJobs;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Newtonsoft.Json.Linq;
 
-namespace Azure.Sdk.Tools.PipelineWitness.Queue.Functions
+namespace Azure.Sdk.Tools.PipelineWitness.Services
 {
-    public class AzurePipelinesBuildCompleteFunction
+    internal class BuildCompleteQueueWorker : QueueWorkerBackgroundService
     {
-        private ILogger logger;
-        private BlobUploadProcessor runProcessor;
+        private readonly ILogger logger;
         private readonly TelemetryClient telemetryClient;
+        private readonly BlobUploadProcessor runProcessor;
 
-        public AzurePipelinesBuildCompleteFunction(ILogger<AzurePipelinesBuildCompleteFunction> logger, BlobUploadProcessor runProcessor, TelemetryClient telemetryClient)
+        public BuildCompleteQueueWorker(
+            ILogger<BuildCompleteQueueWorker> logger,
+            BlobUploadProcessor runProcessor,
+            QueueServiceClient queueServiceClient,
+            TelemetryClient telemetryClient,
+            IOptions<PipelineWitnessSettings> options)
+            : base(
+                logger, 
+                queueServiceClient.GetQueueClient(options.Value.BuildCompleteQueueName), 
```

Comment:
From what I can tell from the docs, this is just a lightweight URI builder? If there's more heavy handed work though, it may be better to build this outside the constructor so errors can be handled more easily.